libfreerdp/core/gateway/rdg.c

```diff
@@ -1098,9 +1098,8 @@ static BOOL rdg_tls_connect(rdpRdg* rdg, rdpTls* tls, const char* peerAddress, i
 
 	if (settings->GatewayPort > UINT16_MAX)
 		return FALSE;
-
-	sockfd = freerdp_tcp_connect(rdg->context, settings, peerAddress ? peerAddress : peerHostname,
-	                             peerPort, timeout);
+	sockfd = rdg->context->update->io->TCPConnect(
```

Comment:
I don´t like the exposure of the internal structure of `rdpTransport` you´re doing here.
We are working on moving all module/components to be self contained with only an opaque structure exposed, this moves in the opposite direction.

Reply:
yes, you right.